src/subscribers.ts

```diff
@@ -4,4 +4,12 @@ export let subscribers: ISubscriber[] = [];
 
 export function clearSubscribers() {
   subscribers = [];
+}
+
+export function removeSubscriberByAction(action: string) {
```

Comment:
Looking at this I was thinking; maybe we shouldn't expose the subscribers array from the this subscribers 'module' as well. So we avoid having the option to manipulate the subscribers array outside of that module. For this we should introduce a `getSubscribers()` function that we export so that is the only way to retrieve the subscribers. Makes things a bit more safe and consistent I think.

Reply:
Same goes for the invokes to keep things consistent.

Reply:
If you want to go this way with it, maybe we could just export a couple of functions that will push to the `subscribers` / `invokes` array?